astropy/io/fits/hdu/hdulist.py

```diff
@@ -820,6 +831,9 @@ def _readfrom(cls, fileobj=None, data=None, mode=None,
 
         saved_compression_enabled = compressed.COMPRESSION_ENABLED
 
+        # (extname, extver) tuple
+        ext = kwargs.pop("ext", None)
```

Comment:
I would suggest renaming "ext" to something that would better show the meaning of this parameter ``stop_reading_data_at_ext``: ``stop_ext``, ``last_read_ext``, etc.